packages/cubejs-api-gateway/src/SubscriptionServer.ts

```diff
@@ -22,12 +24,31 @@ const calcMessageLength = (message: unknown) => Buffer.byteLength(
 export type WebSocketSendMessageFn = (connectionId: string, message: any) => void;
 
 export class SubscriptionServer {
+
+  private cubeRenewSubject = new Subject<any>();
+
   public constructor(
     protected readonly apiGateway: ApiGateway,
     protected readonly sendMessage: WebSocketSendMessageFn,
     protected readonly subscriptionStore: LocalSubscriptionStore,
     protected readonly contextAcceptor: ContextAcceptorFn,
+    protected readonly eventEmitter: EventEmitter
   ) {
+    this.eventEmitter.on('cubeRenewed', (val) => this.cubeRenewSubject.next(val));
+    this.cubeRenewSubject
```

Comment:
Dit soort logica zou ik niet in een constructor stoppen. Deze constructie wil je dan onder je subject definitie zetten en die subscriben.

Reply:
Zeker kan dat wel. Deze pipe constructie wil je dan in je instance variabele stoppen en dan in de constructor subscriben zodat die activeert.